polkadot/runtime/westend/src/lib.rs

```diff
@@ -2237,6 +2240,66 @@ sp_api::impl_runtime_apis! {
 		}
 	}
 
+	impl xcm_fee_payment_runtime_api::dry_run::XcmDryRunApi<Block, RuntimeCall, RuntimeEvent> for Runtime {
+		fn dry_run_extrinsic(extrinsic: <Block as BlockT>::Extrinsic) -> Result<ExtrinsicDryRunEffects<RuntimeEvent>, XcmDryRunApiError> {
+			use xcm_builder::InspectMessageQueues;
```

Comment:
those implementations needs to be duplicated for every runtime? is it possible to somehow abstract it to reduce the amount of duplications?

Reply:
I tried to leave it as a minimum. It's a function call for each element in `XcmDryRunEffects` and some error handling.